sketch/sketch.ino

```diff
@@ -1,363 +1,486 @@
+    Serial.print(isSelected);
+    Serial.print("; clickIndex: ");
+    Serial.print(clickIndex);
+    Serial.print("; chamber: [");
+
+    for (int i = 0; i < 6; i++) {
+      Serial.print(chamber[i]);
+      Serial.print(i != 5 ? "," : "");
+    }
+
+    Serial.println("]");
   }
-  pinMode(buzzerPin, OUTPUT);
-  pinMode(greenButtonPin, INPUT_PULLUP);
-  pinMode(redButtonPin, INPUT_PULLUP);
-  pinMode(knobPin, INPUT);
-}
 
-void loop() {
+  // Copy assignment operator
+  Player& operator=(const Player& other) {
+    // Skip copying if the source is invalid (default constructed)
+    if (other.clickIndex == -1) {
+        Serial.println("Skipping assignment from default-constructed Player.");
+        return *this;
+    }
 
-  switch (currentGameState) {
-    case FirstSetup:
-        // Serial.println("State is FirstSetup");
-        doFirstSetup(playerCount, activePlayers, playerChambers);
-        currentGameState = PlayerIndication;
-      break;
-    case PlayerIndication:
-        // Serial.println("State is PlayerIndication");
-        doPlayerIndication();
-        currentGameState = GameRunning;
-      break;
-    case GameRunning:
-        Serial.println("State is GameRunning");
-        doGameRunning(playerCount, activePlayers);
-        // If input is received to select a player, change state
-        currentGameState = PlayerSelection;
-      break;
-    case PlayerSelection:
-        Serial.println("State is PlayerSelection");
-        doPlayerSelection();
-        currentGameState = PlayerTrigger;
-      break;
-    case PlayerTrigger:
-        Serial.println("State is PlayerTrigger");
-        doPlayerTrigger();
-        currentGameState = GameRunning;
-      break;
-    default:
-        Serial.println("State is defaulting switch block");
-      break;
+    // Free existing memory if already allocated
+    if (chamber != nullptr) {
+        delete[] chamber;
+    }
+
+    // Deep copy the values
+    isAlive = other.isAlive;
+    ledPin = other.ledPin;
+    clickIndex = other.clickIndex;
+    isSelected = other.isSelected;
+
+    chamber = new int[6];
+    for (int i = 0; i < 6; i++) {
+        chamber[i] = other.chamber[i];
+    }
+    return *this;
   }
 
-  delay(50);
-}
+  // Destructor to clean up dynamically allocated memory
+  ~Player() {
+      if (chamber != nullptr) {
+          delete[] chamber;
+      }
+  }
+};
 
-void loadPlayerChambers(int playerChambers[][6], int players) {
-  for (int i = 0; i < players; i++) {
-    // Load bullet at a random place en each chamber
-    playerChambers[i][random(0,6)] = 1;
+// Struct to map LEDs with their state and the player they represent.
+struct PlayerLed {
+  static const int blinkInterval = 200;
+
+  int state;
+  int pin;
+  int playerIndex;
+  unsigned long prevBlinkMillis;
+
+  // Default constructor
+  PlayerLed() : state(-1), pin(-1), playerIndex(-1), prevBlinkMillis(0) { }
+  
+  // Custom constructor
+  PlayerLed(int pinIn, int playerIndexIn) {
```

Comment:
You also can use the 
: state(HIGH),
pin(pinIn),
....
{
}

Reply:
Thanks, not a fan of the syntax, I'll get used to it I guess